handler/handler.go

```diff
@@ -348,3 +349,62 @@ func writeMessage(w http.ResponseWriter, msg string, status int) {
 		log.WithError(err).Error("Failed to parse response message.")
 	}
 }
+
+// ReplaceAnnotation deletes an annotation for a specific content uuid and adds a new one.
+// It gets the annotations only from UPP skipping V2 annotations because they are not editorially curated.
+func (h *Handler) ReplaceAnnotation(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+
+	contentUUID := vestigo.Param(r, "uuid")
+	conceptUUID := vestigo.Param(r, "cuuid")
+
+	tID := tidutils.GetTransactionIDFromRequest(r)
+	ctx := tidutils.TransactionAwareContext(context.Background(), tID)
+
+	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
+	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+	
+	if err := validateUUID(conceptUUID); err != nil {
+		handleWriteErrors("invalid concept UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	conceptUUID = mapper.TransformConceptID("/" + vestigo.Param(r, "cuuid"))
+
+	addedAnnotation := annotations.Annotation{}
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&addedAnnotation)
+	if err != nil {
+		handleWriteErrors("Error decoding request body", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	writeLog.Debug("Validating input and reading annotations from UPP...")
+	uppList, httpStatus, err := h.prepareUPPAnnotations(ctx, contentUUID, addedAnnotation.ConceptId)
+	if err != nil {
+		handleWriteErrors("Error while preparing annotations", err, writeLog, w, httpStatus)
+		return
+	}
+
+	//create a list of annotations with the same ID to keep the predicate
```

Comment:
The following code looks like a major an unnecessary overcomplication of the implementation.
To me the following 5 lines do the same job:
```
	for i, _ := range uppList {
		if uppList[i].ConceptId == conceptID {
			uppList[i].ConceptId = addedAnnotation.ConceptId
		}
	}
```

Reply:
You are right, I fixed it.